src/runtime/scoped_mutex_lock.h

```diff
@@ -10,6 +10,7 @@ struct ScopedMutexLock {
     halide_mutex *mutex;
 
     ScopedMutexLock(halide_mutex *mutex) __attribute__((always_inline)) : mutex(mutex) {
+        halide_mutex_init(mutex);
```

Comment:
Can we not do this? This affects every platform. If this is truly necessary, a better approach might be to just make it so the QuRT implementation of halide_mutex_lock just initializes if it wasn't already.

Reply:
Their previous implementation was in fact based on doing the said check inside the QuRT implementation of halide_mutex_lock. However, there wasn't an atomic thread-safe way of doing that and that was the problem.

Reply:
Is this any better though? I don't see how this is different in terms of thread safety.